lmdeploy/pytorch/engine/engine.py

```diff
@@ -207,9 +264,7 @@ def _bind_request_manager(self):
 
     def _start_loop(self):
         """start loop."""
-        loop_threads = Thread(target=self.loop, daemon=True)
-        loop_threads.start()
-        return loop_threads
+        return self.req_manager.start_loop(self.async_loop)
```

Comment:
failed with opencompass inferrence:
```
2024-03-01 14:15:32,718 - lmdeploy - INFO - build CacheEngine with config:CacheConfig(block_size=64, num_cpu_blocks=128, num_gpu_blocks=1678, window_size=-1, cache_max_entry_count=0.8)
03/01 14:15:34 - OpenCompass - INFO - Start inferencing [Llama-2-7b-chat-hf/gsm8k_0]
[2024-03-01 14:15:35,194] [opencompass.openicl.icl_inferencer.icl_gen_inferencer] [INFO] Starting inference process...
^M  0%|          | 0/12 [00:00<?, ?it/s]^M  0%|          | 0/12 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/opt/opencompass/opencompass/tasks/openicl_infer.py", line 153, in <module>
    inferencer.run()
  File "/opt/opencompass/opencompass/tasks/openicl_infer.py", line 81, in run
    self._inference()
  File "/opt/opencompass/opencompass/tasks/openicl_infer.py", line 126, in _inference
    inferencer.inference(retriever,
  File "/opt/opencompass/opencompass/openicl/icl_inferencer/icl_gen_inferencer.py", line 146, in inference
    results = self.model.generate_from_template(
  File "/opt/opencompass/opencompass/models/base.py", line 165, in generate_from_template
    return self.generate(inputs, max_out_len=max_out_len, **kwargs)
  File "/opt/opencompass/opencompass/models/lmdeploy_pytorch.py", line 97, in generate
    _results = list(
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 619, in result_iterator
    yield fs.pop().result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/opencompass/opencompass/models/lmdeploy_pytorch.py", line 149, in _generate
    generator.end(session_id)
  File "/opt/lmdeploy/lmdeploy/pytorch/engine/engine.py", line 1172, in end
    return end(self.req_sender, session_id)
  File "/opt/lmdeploy/lmdeploy/pytorch/engine/engine.py", line 135, in end
    resp = req_sender.send(RequestType.END_SESSION,
  File "/opt/lmdeploy/lmdeploy/pytorch/engine/request.py", line 345, in send
    req_id = self.send_async(req_type, data)
  File "/opt/lmdeploy/lmdeploy/pytorch/engine/request.py", line 273, in send_async
    return self.batched_send_async(req_types=[req_type], data=[data])[0]
  File "/opt/lmdeploy/lmdeploy/pytorch/engine/request.py", line 265, in batched_send_async
    return asyncio.get_event_loop().run_until_complete(coro)
  File "/usr/lib/python3.8/asyncio/events.py", line 639, in get_event_loop
    raise RuntimeError('There is no current event loop in thread %r.'
RuntimeError: There is no current event loop in thread 'ThreadPoolExecutor-0_0'.
sys:1: RuntimeWarning: coroutine 'RequestSender.async_batched_send_async' was never awaited
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

Reply:
Fixed, if it still not working, try enable `thread_safe` flag in `PyTorchEngineConfig`

Reply:
@RunningLeon Let's prioritize our inference performance. Users are very sensitive to it.

Reply:
ok. we may need to PR to opencompass once this PR is merged.